src/libertem/io/dataset/base/tiling.py

```diff
@@ -511,6 +523,22 @@ def get_scheme(self, udfs, partition, read_dtype: np.dtype, roi: np.ndarray):
 
         # the partition has a "veto" on the tileshape:
         tileshape = partition.adjust_tileshape(tileshape)
+        # The correction has to make sure that there are no excluded pixels
+        # at tile boundaries
+        tileshape_corrected = corrections.adjust_tileshape(tileshape, tuple(partition.shape.sig))
```

Comment:
Hmm, I think this may need further tweaking - `adjust_tileshape` assumes that the negotiation did take the `base_shape` into account, so for example k2is doesn't adjust the tileshape back to something that is divisible by `(1, 930, 16)`.

`adjust_tileshape` is just a blunt instrument to make formats like HDF5 and SER work, where we don't necessarily have direct control over how the data is read (HDF5 ds could be improved to work better with chunking, but that's another story)

Reply:
👍